ddtrace/llmobs/_writer.py

```diff
@@ -343,6 +344,8 @@ def dataset_pull(self, name: str) -> Dataset:
         if not data:
             raise ValueError(f"Dataset '{name}' not found")
 
+        curr_version = data[0]["attributes"]["current_version"]
```

Comment:
should we be safe about len of `data` and its objects within

Reply:
I'm following Kyle's pattern of explicitly failing for unexpected values (see https://github.com/DataDog/dd-trace-py/pull/13852#discussion_r2190777700). We can move this to a follow up to be more safe (regardless we'll still be raising if the response data is not the expected format)

Reply:
>  but to truly support versioning we need delta tracking

I assume you're talking about tracking implicit version changes after dataset batch/update calls right? I'll handle that once #13951 is merged